rule/expr.go

```diff
@@ -241,7 +238,13 @@ func extractPlatformStringsExprs(expr bzl.Expr) (platformStringsExprs, error) {
 		switch part := part.(type) {
 		case *bzl.ListExpr:
 			if ps.generic != nil {
-				return platformStringsExprs{}, fmt.Errorf("expression could not be matched: multiple list expressions")
+				return platformStringsExprs{}, fmt.Errorf("expression could not be matched: multiple generic expressions")
+			}
+			ps.generic = part
+
+		case *bzl.DictExpr:
```

Comment:
A bare dict cannot be part of `platformStringsExpr`. This function matches a specific pattern, and that shouldn't change.